src/parser/Tokenizer.cc

```diff
@@ -168,6 +168,17 @@ Parser::Tokenizer::skipOne(const CharacterSet &chars)
     return false;
 }
 
+char
+Parser::Tokenizer::skipOne()
+{
+    if (buf_.isEmpty())
+        throw InsufficientInput();
+
+    debugs(24, 8, "skipping one of any character");
+    const auto result = consume(1);
+    return result[0];
```

Comment:
I doubt the compiler will be able to optimize the creation of a temporary SBuf away in this case. I would help the compiler by doing something like this instead:

```suggestion
    const auto result = buf_[0];
    consume(1);
    debugs(24, 8, "character code: " << int(result));
    return result;
```

N.B. debugs() will print "skipOne" so we do not need to say "skipping one" or some such.